### ISSUE TITLE
Terraform Destroy crashed 

### ISSUE THREAD

**Issue by VincentSastra** [hashicorp/terraform-provider-aws#20878]
Version 3.58

```
terraform apply -destroy

╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).UpgradeResourceState call. The plugin    
│ logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.image-storage.aws_api_gateway_method.get-image,
│   on image-storage\image-storage.tf line 22, in resource "aws_api_gateway_method" "get-image":
│   22: resource "aws_api_gateway_method" "get-image" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may   
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.image-storage.aws_api_gateway_integration.get-image,
│   on image-storage\image-storage.tf line 29, in resource "aws_api_gateway_integration" "get-image":
│   29: resource "aws_api_gateway_integration" "get-image" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).UpgradeResourceState call. The plugin    
│ logs may contain more details.
╵
```

```
Stack trace from the terraform-provider-aws_v3.58.0_x5.exe plugin:

panic: interface conversion: interface {} is nil, not *aws.AWSClient

goroutine 84 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsApiGatewayMethodRead(0xc0017da080, 0x0, 0x0, 0xc00178d620, 
0xc000100400)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_api_gateway_method.go:164 +0xa99
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000abce00, 0x7fbc088, 0xc001783540, 0xc0017da080, 0x0, 0x0, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/gdavison/terraform-plugin-sdk/v2@v2.0.2-0.20210714181518-b5a3dc95a675/helper/schema/resource.go:335 +0x1fb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000abce00, 0x7fbc088, 0xc001783540, 0xc001795ea0, 0x0, 0x0, 0xc000da4b20, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/gdavison/terraform-plugin-sdk/v2@v2.0.2-0.20210714181518-b5a3dc95a675/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000004ed0, 0x7fbc088, 0xc001783540, 0xc001783580, 0xc001783540, 0x15beca, 0x6ed9400)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/gdavison/terraform-plugin-sdk/v2@v2.0.2-0.20210714181518-b5a3dc95a675/helper/schema/grpc_provider.go:575 +0x45b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc001ff6280, 0x7fbc130, 0xc001783540, 0xc001798780, 0xc001ff6280, 0xc002385680, 0xc002549ba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x108
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x71c7000, 0xc001ff6280, 0x7fbc130, 0xc002385680, 0xc001798720, 0x0, 0x7fbc130, 0xc002385680, 0xc001770120, 0x115)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x222
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000241a40, 0x7fdeff8, 0xc001218180, 0xc000376400, 0xc0018e8ae0, 0xb992650, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000241a40, 0x7fdeff8, 0xc001218180, 0xc000376400, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0001461b0, 0xc000241a40, 0x7fdeff8, 0xc001218180, 0xc000376400)        
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-aws_v3.58.0_x5.exe plugin crashed!
```

**Comment by ewbankkit** [comment]
Similar:

* #20461
* #20438
* #20131
* #20015
* #19807
* #19788
* #19740
* #19601

**Comment by gdavison** [comment]
Hi @VincentSastra, what method of authenticating with AWS are you using? In some cases, we see the error `panic: interface conversion: interface {} is nil, not *aws.AWSClient` when there is an authentication failure with AWS.

Is this a consistent error, or does it only occur occasionally?

**Comment by justinretzolk** [comment]
👋 The other reports we've had of this issue were all resolved by Terraform `1.0.9`; since we haven't heard back on this issue, I'll assume the same and close this one out. If you run into additional issues, please open a new issue so that we may continue to investigate.
